pkg/sriov/utils/utils_provider.go

```diff
@@ -0,0 +1,229 @@
+	if err != nil {
+		return 0, errors.Errorf("unable to convert string to int from sriov_numvfs file for device %s: %v", pfPciAddr, err)
+	}
+	return numConfiguredVFs, nil
+}
+
+func (u *sriovUtilsProvider) IsSriovConfigured(pciAddr string) bool {
+	numVfs, err := u.GetConfiguredVirtualFunctionsNumber(pciAddr)
+	if err != nil {
+		return false
+	}
+	return numVfs > 0
+}
+
+func (u *sriovUtilsProvider) GetSriovVirtualFunctionsCapacity(pfPciAddr string) (int, error) {
+	totalVfFilePath := filepath.Join(u.sysfsDevicesPath, pfPciAddr, totalVfFile)
+	vfs, err := ioutil.ReadFile(filepath.Clean(totalVfFilePath))
+	if err != nil {
+		return 0, errors.Errorf("unable to locate sriov_totalvfs file for device %s: %v", pfPciAddr, err)
+	}
+	totalVfs := bytes.TrimSpace(vfs)
+	numTotalVfs, err := strconv.Atoi(string(totalVfs))
+	if err != nil {
+		return 0, errors.Errorf("unable to convert string to int from sriov_totalvfs file for device %s: %v", pfPciAddr, err)
+	}
+	return numTotalVfs, nil
+}
+
+func (u *sriovUtilsProvider) IsDeviceExists(pciAddr string) (bool, error) {
+	// init sysfs pci address regex
+	initRegexpOnce.Do(func() {
+		validLongPCIAddr = regexp.MustCompile(`^0{4}:[0-9a-f]{2}:[0-9a-f]{2}.[0-7]{1}$`)
+		validShortPCIAddr = regexp.MustCompile(`^[0-9a-f]{2}:[0-9a-f]{2}.[0-7]{1}$`)
+	})
+
+	// Check system pci address
+	if validLongPCIAddr.MatchString(pciAddr) {
+		devicePath := filepath.Join(u.sysfsDevicesPath, pciAddr)
+		return u.isFileOrLinkExists(devicePath), nil
+	} else if validShortPCIAddr.MatchString(pciAddr) {
+		pciAddr = bdfDomain + pciAddr // convert short form to sysfs' long form representation
+		devicePath := filepath.Join(u.sysfsDevicesPath, pciAddr)
+		return u.isFileOrLinkExists(devicePath), nil
+	}
+	return false, errors.Errorf("invalid pci address provided: %s", pciAddr)
+}
+
+func (u *sriovUtilsProvider) GetNetInterfacesNames(pciAddr string) ([]string, error) {
+	var names []string
+	netDir := filepath.Join(u.sysfsDevicesPath, pciAddr, netInterfacesPath)
+	if _, err := os.Lstat(netDir); err != nil {
+		return nil, errors.Errorf("no net directory under pci device %s: %v", pciAddr, err)
+	}
+
+	fInfos, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return nil, errors.Errorf("failed to read net directory %s: %v", netDir, err)
+	}
+
+	names = make([]string, 0)
+	for _, f := range fInfos {
+		names = append(names, f.Name())
+	}
+	return names, nil
+}
+
+func (u *sriovUtilsProvider) CreateVirtualFunctions(pfPciAddr string, vfNumber int) error {
+	if vfNumber < 1 {
+		return errors.Errorf("invalid number of virtual functions specified: %d. Must be > 0", vfNumber)
+	}
+	if u.IsSriovConfigured(pfPciAddr) {
+		return errors.Errorf("virtual functions are already exist for device: %s", pfPciAddr)
+	}
+	configuredVfPath := filepath.Join(u.sysfsDevicesPath, pfPciAddr, configuredVfFile)
+	err := ioutil.WriteFile(configuredVfPath, []byte(strconv.FormatInt(int64(vfNumber), 10)), 0600)
```

Comment:
Can we use this constat here?
```suggestion
	err := ioutil.WriteFile(configuredVfPath, []byte(strconv.FormatInt(int64(vfNumber), 10)), os.ModePerm)
```

Reply:
`os.ModePerm` = 0777. When i write 0777 by myself without this os constant, golangci-lint complains:
> G306: Expect WriteFile permissions to be 0600 or less (gosec)

But when i use `os.ModePerm` golangci-lint says it's OK. Looks like linter isn't very smart to detect that it is the same 0777 is complains about.
So, i think we should use less permissive bits.